cdap-security/src/main/java/io/cdap/cdap/security/tools/TokenGeneratorTool.java

```diff
@@ -0,0 +1,120 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.tools;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.guice.InMemoryDiscoveryModule;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.security.auth.AbstractKeyManager;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.FileBasedKeyManager;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+
+/**
+ * A command line tool for generating access tokens.
+ */
+public class TokenGeneratorTool {
+
+  private void doMain(String[] args) throws ParseException, IOException {
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
+      .addOption(new Option("u", "username", true, "Username/principal"));
+
+    CommandLineParser parser = new BasicParser();
+    CommandLine commandLine = parser.parse(options, args);
+
+    if (commandLine.hasOption("h") || commandLine.getOptions().length == 0) {
+      printUsage(options);
+      return;
+    }
+
+    if (commandLine.hasOption("u")) {
+      String username = commandLine.getOptionValue("u");
+      printAccessToken(username);
+      return;
+    }
+
+    printUsage(options);
+  }
+
+  @VisibleForTesting
+  void printAccessToken(String username) throws IOException {
+    // Suppress logging from CDAP classes
+    ((Logger) LoggerFactory.getLogger(TokenManager.class)).setLevel(Level.ERROR);
+    ((Logger) LoggerFactory.getLogger(FileBasedKeyManager.class)).setLevel(Level.ERROR);
+    ((Logger) LoggerFactory.getLogger(AbstractKeyManager.class)).setLevel(Level.ERROR);
+
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(
+      new ConfigModule(cConf),
+      new IOModule(),
+      new InMemoryDiscoveryModule(),
+      CoreSecurityRuntimeModule.getDistributedModule(cConf));
```

Comment:
By default, this will return the InMemorySecurityModule which will generate a random key every time which seems like it should not be intended behavior. Instead of relying on a mounted CConf, can we specify a command-line option to determine which KeyManager module is used? If none is specified, we should default to file-based key manager.

Reply:
I can set security.enabled = true to ensure that FileBasedKeyManager is used

Reply:
fixed